pkgs/os-specific/linux/trace-cmd/kernelshark.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv, mkDerivation, fetchgit, qtbase, cmake, json_c, mesa_glu, freeglut, trace-cmd }:
+let
+  srcSpec = import ./src.nix;
+  shortRev = builtins.substring 0 7 srcSpec.rev;
+in mkDerivation rec {
+  pname = "kernelshark";
+  version = "0.9.8-${shortRev}";
+
+  src = fetchgit srcSpec;
+
+  sourceRoot = "trace-cmd-${shortRev}/kernel-shark";
+
+  patches = [ ./fix-kernel-shark-CMakeLists.txt.patch ];
+  patchFlags = [ "-p2" ];
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ qtbase json_c mesa_glu freeglut ];
+
+  cmakeFlags = [
+    "-DTRACECMD_BIN_DIR=${trace-cmd}/bin"
+    "-DTRACECMD_INCLUDE_DIR=${trace-cmd.dev}/include/trace-cmd"
+    "-DTRACECMD_LIBRARY_DIR=${trace-cmd.lib}/lib"
+    "-DTRACEEVENT_INCLUDE_DIR=${trace-cmd.dev}/include/trace-cmd"
+    "-DTRACEEVENT_LIBRARY_DIR=${trace-cmd.lib}/lib"
+  ];
+
+  preConfigure = ''cmakeFlagsArray=($cmakeFlagsArray "-D_INSTALL_PREFIX=$out")'';
```

Comment:
You should use `"-D_INSTALL_PREFIX=${placeholder "out"}"`

Reply:
That's much nicer indeed. Will adapt.